### PR TITLE
Fixed #25, etc

### LINKED ISSUE #25: [SIM800] WaitforNetwork doesn't actually wait for a network connection
When I run waitForNetwork(), I expect that it will and keep trying until a connection is made.  But it seems to be making one attempt to get the network status and then just returning false.  In the code, I think the for loop of waitForNetwork() should actually be a while loop instead.  Am I doing something wrong?

```cpp
Serial.println(F("Waiting for network..."));  // For debugging
if (!_modem->waitForNetwork()){
  Serial.println("... Connection failed");  // For debugging
} else {
   _modem->gprsConnect(_APN, "", "");
}
```

### PR DESCRIPTION
**Pull request by SRGDamia1**
Fixed #25 by removing extra return statement so WaitForNetwork() really waits.

Moved all of the phone/sms/location functions together (there were two groups of each).

Made sendAT and WaitforResponse both public so it's easy for users to send their own less commonly used commands.  (Could be helpful for requests like #5.)

Added some extra A's to SIM800's autoBaud detection because mine wasn't detecting the baud rate with only the one in the AT command.

Made streamSkipUntil dependent on stream.readStringUntil(c), which has a built-in timeout.

Created a streamReadUntil function, mostly to help myself with debugging.

Added a few other debugging bits.

### CODE REVIEW

**Comment by vshymanskyy** [comment]
Hi! Thank you for a nice PR. I really appreciate your efforts.
Could you please elaborate on XBee support that you added?

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,9 @@
 # Extras
 extras/docs/*
 
+.pioenvs
```
Comment (vshymanskyy) [inline .gitignore]: Do you think we need to add these exclusions to the main repo?

Comment (SRGDamia1) [inline .gitignore]: I'm sorry; probably not.  I added them for myself.

**Comment by SRGDamia1** [comment]
The Digi XBee support makes an easier interface (ie, unified with your other modems) for getting in and out of their "command mode," which is necessary for changing networks and opening/closing sockets if you don't want to use their complicated API frames.  Because the XBee's already operate in a transparent mode, the FIFO isn't needed to turn them into proper clients.

**Comment by SRGDamia1** [comment]
Mostly, I loved what you've done to simplify using TCP/IP on the SIM800 and wanted to make it easier to switch between my GPRSBees and XBee S6B's.

**Comment by vshymanskyy** [comment]
Ok, thanks. I have 2 XBee modules so I will hopefully be able to test it (let's discuss this a bit later).
For now, let's finish reviewing this PR - I would love to include these changes.
Thanks again.

**Comment by vshymanskyy** [comment]
@SRGDamia1 Let me invite you to or new TinyGSM developer chat:
https://gitter.im/tinygsm/Developers?utm_source=share-link&utm_medium=link&utm_campaign=share-link

**Comment by vshymanskyy** [comment]
@SRGDamia1 The fix was incorporated, and used for other modules as well.
I also rearranged the functions so it makes more sense and is much easier to compare with your version.
Please review your other changes, and send a separate PR with other valuable stuff.
For XBee support - this is really interesting, but i would like to test it with my hardware (i have some) before merging this in. Could you help me setting up such a test? Thank you!

**Pull request closed without merging**